crypto_kem/kyber768/clean/GNUmakefile

```diff
@@ -0,0 +1,15 @@
+LIB=libkyber768_clean.a
```

Comment:
This file is named `GNUmakefile`. Are we using any GNU-specific things, or should this just be `Makefile`? There are platforms that won't necessarily have GNU Make (BSD comes to mind).

Reply:
I am not sure, I guess I could install a BSD VM to test.  Is BSD a targetted platform?

The reason I called it GNUmakefile is that apparently GNU Make will search filenames in the following order: GNUmakefile, makefile, Makefile.  Whereas Windows nmake will search for MAKEFILE (case insensitive) and then give up.  If I called this one "Makefile", then Windows nmake would see it and fail, whereas if I call it GNUmakefile, then GNU Make will still see it, but Windows nmake won't.

Reply:
OS X is a BSD flavour, though I'm also just making assumptions here. It seems like lowhanging fruit to be more inclusive.

The Windows thing may be a  good reason to name it `GNUMakefile`, or we could just rely on nmake respecting `/f` (which I assume makes it skip this search) and accept that the (probably a minority) of Windows devs will just have to specify it (or throw away the Unix Makefile).

Reply:
Or I could just install BSD Make on my Mac.  Which I have done.  bsdmake fails with our GNUmakefile: the following two lines don't work:

    SOURCES=$(wildcard *.c)
    OBJECTS=$(patsubst %.c,%.o,$(SOURCES))

If I explicitly list the .o files in the makefile (like I had to do for Makefile.Microsoft_nmake) then we could have the same makefile to work for both GNU Make and BSD Make.

Reply:
Ah, I had no idea that these were GNU extensions. Seems too annoying to work without them though, so maybe we can also trust that BSD people know how to solve this problem :man_shrugging:.

Reply:
Well actually it's not that annoying to just list the .o files by name, and then it becomes an even more obvious makefile, so I've made that change, and then it's good for both GNU and BSD Make, and probably other random Makes out there.  Once Travis CI passes I'll merge it.